Source/MQTTnet/Implementations/CrossPlatformSocket.cs

```diff
@@ -23,7 +23,7 @@ public CrossPlatformSocket()
         {
             // Having this constructor is important because avoiding the address family as parameter
             // will make use of dual mode in the .net framework.
-            _socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+            _socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
Instead of making a change here you can set the address family to something different than "AddressFamily.Unspecified" in the client options. Then the code will use the other constructor.